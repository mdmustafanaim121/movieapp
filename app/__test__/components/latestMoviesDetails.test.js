import LatestMoviesDetails from '../../components/latestMoviesDetails';
import React from 'react';
import renderer from 'react-test-renderer';


describe('Testing Latest Movies', () => {
    it('Should return latest movies snapshot',()=>{
       const route={params:
            {
            name:'Black Panther',
            rate:'2', 
            type: 'https://image.ibb.co/f0hhZc/bp.jpg', 
            language:'english'}
       }
        const result = renderer.create(<LatestMoviesDetails route={route}/>).toJSON()
        expect(result).toMatchSnapshot()
        })
})
