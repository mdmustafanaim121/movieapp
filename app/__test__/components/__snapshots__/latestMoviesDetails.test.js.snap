// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Latest Movies Should return latest movies snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#e1e8ee",
            "borderWidth": 1,
            "margin": 15,
            "marginBottom": 0,
            "padding": 15,
            "shadowColor": "rgba(0,0,0, .2)",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
                "resizeMode": "cover",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <Text
              style={
                Object {
                  "color": "#43484d",
                  "fontSize": 28,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                  "textAlign": "center",
                }
              }
              testID="cardTitle"
            >
              Black Panther
            </Text>
            <View
              style={
                Object {
                  "borderBottomColor": "#bcbbc1",
                  "borderBottomWidth": 0.5,
                  "marginBottom": 15,
                }
              }
            />
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "display": "flex",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "overflow": "hidden",
                      "position": "relative",
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    source={
                      Object {
                        "uri": undefined,
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "height": 300,
                        "left": 0,
                        "marginBottom": 20,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": 200,
                      }
                    }
                    testID="RNE__Image"
                    theme={
                      Object {
                        "colors": Object {
                          "black": "#242424",
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "grey": "rgba(0, 0, 0, 0.54)",
                              "primary": "#2196f3",
                              "searchBg": "#dcdce1",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "default": Object {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "web": Object {
                              "error": "#ff190c",
                              "grey": "#393e42",
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#ca71eb",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                          "white": "#ffffff",
                        },
                      }
                    }
                    transition={true}
                    transitionDuration={360}
                  />
                  <View
                    accessibilityElementsHidden={true}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#bdbdbd",
                          "height": 300,
                          "justifyContent": "center",
                          "marginBottom": 20,
                          "width": 200,
                        }
                      }
                      testID="RNE__Image__placeholder"
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "height": 300,
                        "marginBottom": 20,
                        "width": 200,
                      }
                    }
                    testID="RNE__Image__children__container"
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "borderBottomColor": "#bcbbc1",
                    "borderBottomWidth": 0.5,
                    "marginBottom": 15,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "display": "flex",
                    "flexDirection": "column",
                    "marginBottom": 10,
                    "paddingLeft": 10,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "marginBottom": 10,
                        "marginRight": 10,
                      }
                    }
                  >
                    Type:
                  </Text>
                  <Text
                    style={
                      Object {
                        "marginBottom": 10,
                      }
                    }
                  >
                    https://image.ibb.co/f0hhZc/bp.jpg
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "marginBottom": 10,
                        "marginRight": 10,
                      }
                    }
                  >
                    Language:
                  </Text>
                  <Text
                    style={
                      Object {
                        "marginBottom": 10,
                      }
                    }
                  >
                    english
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "marginRight": 10,
                        "marginTop": 4,
                      }
                    }
                  >
                    Ratings:
                  </Text>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "flexDirection": "column",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            Object {
                              "height": 20,
                              "margin": 3,
                              "tintColor": "#f1c40f",
                              "transform": Array [
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "width": 20,
                            }
                          }
                        />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            Object {
                              "height": 20,
                              "margin": 3,
                              "tintColor": "#f1c40f",
                              "transform": Array [
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "width": 20,
                            }
                          }
                        />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            Object {
                              "height": 20,
                              "margin": 3,
                              "tintColor": "#BDC3C7",
                              "transform": Array [
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "width": 20,
                            }
                          }
                        />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            Object {
                              "height": 20,
                              "margin": 3,
                              "tintColor": "#BDC3C7",
                              "transform": Array [
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "width": 20,
                            }
                          }
                        />
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            Object {
                              "height": 20,
                              "margin": 3,
                              "tintColor": "#BDC3C7",
                              "transform": Array [
                                Object {
                                  "scale": 1,
                                },
                              ],
                              "width": 20,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "marginRight": 10,
                        "marginTop": 4,
                      }
                    }
                  >
                    No of Seats:
                  </Text>
                  <View
                    style={
                      Object {
                        "height": 20,
                        "width": 150,
                      }
                    }
                  >
                    <RCTPicker
                      items={
                        Array [
                          Object {
                            "label": "1",
                            "textColor": undefined,
                            "value": "1",
                          },
                          Object {
                            "label": "2",
                            "textColor": undefined,
                            "value": "2",
                          },
                          Object {
                            "label": "3",
                            "textColor": undefined,
                            "value": "3",
                          },
                        ]
                      }
                      onChange={[Function]}
                      selectedIndex={0}
                      style={
                        Array [
                          Object {
                            "height": 216,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "marginRight": 10,
                        "marginTop": 4,
                      }
                    }
                  >
                    Date:
                  </Text>
                  <View
                    style={
                      Object {
                        "height": 20,
                        "width": 200,
                      }
                    }
                  >
                    <RCTPicker
                      items={
                        Array [
                          Object {
                            "label": "07th July 2021",
                            "textColor": undefined,
                            "value": "07th July 2021",
                          },
                          Object {
                            "label": "21st August 2021",
                            "textColor": undefined,
                            "value": "21st August 2021",
                          },
                          Object {
                            "label": "19th January 2021",
                            "textColor": undefined,
                            "value": "19th January 2021",
                          },
                        ]
                      }
                      onChange={[Function]}
                      selectedIndex={0}
                      style={
                        Array [
                          Object {
                            "height": 216,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "marginRight": 10,
                        "marginTop": 4,
                      }
                    }
                  >
                    Time:
                  </Text>
                  <View
                    style={
                      Object {
                        "height": 20,
                        "width": 150,
                      }
                    }
                  >
                    <RCTPicker
                      items={
                        Array [
                          Object {
                            "label": "7:30 pm",
                            "textColor": undefined,
                            "value": "7:30 pm",
                          },
                          Object {
                            "label": "1:30 pm",
                            "textColor": undefined,
                            "value": "1:30 pm",
                          },
                          Object {
                            "label": "5:00 pm",
                            "textColor": undefined,
                            "value": "5:00 pm",
                          },
                        ]
                      }
                      onChange={[Function]}
                      selectedIndex={0}
                      style={
                        Array [
                          Object {
                            "height": 216,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": "#2089dc",
                      "borderColor": "#2089dc",
                      "borderRadius": 8,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginBottom": 10,
                      "marginTop": 20,
                      "padding": 8,
                      "width": "50%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Book Now
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
